### PR TITLE
feat: ses email setup

### PR DESCRIPTION
**Pull request by emilyjablonski**
# Pull Request Template

## Issue Overview

This PR addresses [323](https://github.com/metrotranscom/doorway/issues/323)

- [ ] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

Notes

* The reason that we're using nodemailer is because of our potential need to send emails with attachments. Out of the box, the SES sendEmail function does everything we need it to do, but they don't have a nice way to add attachments to emails. To add an attachment, you need to use sendRawEmail, which makes you build a MIME formatted email from scratch, which is icky - nodemailer wraps the SES API and allows us to do that more easily. Even though we only have one potential need for the CSV export attachment, I'm using nodemailer everywhere for consistency.
* SES has a system for managing templates and using templated data in handlebars files which is what I tried initially for our emails, but there isn't a great system for creating / updating / managing the templates in SES. You need to hit the API directly, and there isn't a good way of managing them in the AWS UI. It would be somewhat of a pain to need to store handlebars files, convert them to strings, and send em to SES whenever we want to make a string change. I'm continuing the same pattern we were using before, which is that we just smush the data we need into an already compiled handlebars template and send all of that as the email HTML, without actually using templated data. While templates may offer some nice benefits, an additional problem was that if you have any malformatted or missing template data, SES will not send the email, but also intentionally won't send you back a failure in the API - it will silently fail. The failure indication would only be available in CloudWatch logs which is a bummer.
* There is no longer a need in Doorway to have an emailFromAddress as everything will be no-reply@housingbayarea.org
* Logs for our SES account are manually routed to a dashboard in CloudWatch. There are additional tags we can add to emails for CloudWatch if we want to be able to identify more details about certain email metrics. It's kind of hard to tell without more data which forms of displaying the data are going to be the most useful. Understanding which tags we want might be iterative as we have different queries of log data. At the moment we have data on sends / fails.

## How Can This Be Tested/Reviewed?

Send emails locally and ensure they come through and are formatted correctly.

Remaining questions
* Do we want the same retry functionality we had for SendGrid?
* Do we want to use nodemailer's built-in rate limiting?

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: api/src/views/partials/header.hbs [hunk 1]
```diff
@@ -13,7 +13,7 @@
                     <tbody>
                       <tr>
                         <td align="center">
-                          <img src="{{t "header.logoUrl"}}" alt="{{t "header.logoTitle"}}" width="300" height="65" />
+                          <img src="{{t "header.logoUrl"}}" alt="{{t "header.logoTitle"}}" width="300" height="65" class="header-image"/>
```
Comment (emilyjablonski) [inline api/src/views/partials/header.hbs:16]: This was an attempt to make the logo readable in dark mode

File: api/src/services/email.service.ts [hunk 2]
```diff
@@ -23,33 +23,34 @@ import {
...
-  type: string;
-};
-
 type listingInfo = {
   id: string;
   name: string;
   juris: string;
 };
 
+const sesClient = new aws.SESClient({
```
Comment (emilyjablonski) [inline api/src/services/email.service.ts:41]: Curious where in this file you think these should live? @YazeedLoonat

File: api/src/services/email.service.ts [hunk 3]
```diff
@@ -143,46 +144,32 @@ export class EmailService {
...
+  public async sendSES(mailOptions: SendMailOptions) {
+    try {
+      return await transporter.sendMail({
+        ...mailOptions,
+        from: 'Doorway <no-reply@housingbayarea.org',
+      });
+    } catch (e) {
+      console.log(e);
+      console.error('Failed to send email');
+      // todo: do we want this sendgrid retry behavior?
```
Comment (emilyjablonski) [inline api/src/services/email.service.ts:156]: @YazeedLoonat Curious for your thoughts here

File: api/src/services/email.service.ts (hunk 2 above)
Comment (YazeedLoonat) [reply api/src/services/email.service.ts:41]: thats a great question, I think moving this and the below `const transporter = ...` into the constructor() makes the most sense

And then make `transporter` an instance variable like `polyglot`

**Review by YazeedLoonat — CHANGES_REQUESTED** [review]
Looks good a few notes in response to your comments!

For the question of `Do we want to use nodemailer's built-in rate limiting?` I don't think we do. Since we send at least 1 email per application submission, the numbers can rack-up quickly. We should rely on the rate limiter to our API I think

File: api/src/services/email.service.ts [hunk 4]
```diff
@@ -143,46 +146,17 @@ export class EmailService {
...
+  public async sendSES(mailOptions: SendMailOptions) {
+    try {
+      return await this.transporter.sendMail({
+        ...mailOptions,
+        from: 'Doorway <no-reply@housingbayarea.org>',
+      });
+    } catch (e) {
+      console.log(e);
+      console.error('Failed to send email');
+      return e;
```
Comment (YazeedLoonat) [inline api/src/services/email.service.ts:158]: question: wondering if instead returning an error that we generate would be better? 

maybe throwing `InternalServerErrorException` ?

**Review by YazeedLoonat — APPROVED** [review]
LGTM, small question but not blocking

File: api/src/services/email.service.ts (hunk 4 above)
Comment (emilyjablonski) [reply api/src/services/email.service.ts]: updated ty!

**Review by ColinBuyck — APPROVED** [review]
Sweetttt, LGTM *️⃣! Only thing is could you update the branch and just ensure the confirmation code is coming through on the application confirmation email?

**Pull request merged**